cpp/src/io/statistics/column_stats.cu

```diff
@@ -165,9 +165,15 @@ gatherIntColumnStats(stats_state_s *s, statistics_dtype dtype, uint32_t t, Stora
     uint32_t r                = i + t;
     uint32_t row              = r + s->group.start_row;
     const uint32_t *valid_map = s->col.valid_map_base;
-    uint32_t is_valid         = (r < s->group.num_rows && row < s->col.num_values)
-                          ? (valid_map) ? (valid_map[row >> 5] >> (row & 0x1f)) & 1 : 1
-                          : 0;
+    uint32_t is_valid         = 0;
+    if (r < s->group.num_rows && row < s->col.num_values) {
+      if (valid_map != nullptr) {
+        is_valid =
+          (valid_map[(row + s->col.column_offset) / 32] >> ((row + s->col.column_offset) % 32)) & 1;
+      } else {
+        is_valid = 1;
+      }
+    }
```

Comment:
My comment was referring to all instances of this code (occurs 6 time AFAIK). IMO it would be better to keep the code consistent, rather than change one of the instances. We can clean this up in a separate PR, given that this one is a bug fix.